packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -231,6 +232,7 @@ export abstract class RetrieveExecutor<T> extends DeployRetrieveExecutor<T> {
       const relativePackageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
       const output = this.createOutput(result, relativePackageDirs);
       channelService.appendLine(output);
+      PersistentStorageService.getInstance().setPropertiesForFiles(result.response.fileProperties);
```

Comment:
I believe that fileProperties will contain the information for files that were successfully retrieved and only for files that were successfully retrieved. If that is not true, can perhaps add an if statement here.

Reply:
Is a check necessary? It's not as long as there are no extra fileProperties in the result right

Reply:
We discussed offline and decided not to have this check.

Reply:
Doesn't need to be part of this PR, but in a future one we could consider exporting the instance of the PersistentStorageService similar to what we do with telemetry [here](https://github.com/forcedotcom/salesforcedx-vscode/blob/7a8858b60cca1534197f04e764cbbcca2edfed53/packages/salesforcedx-vscode-core/src/telemetry/index.ts#L11)